src/types/timestamp.rs

```diff
@@ -1584,6 +1644,114 @@ mod timestamp_tests {
         Ok(())
     }
 
+    #[test]
+    fn test_timestamp_year() -> IonResult<()> {
+        let timestamp = Timestamp::with_year(2021).with_month(2).build()?;
+        assert_eq!(timestamp.year(), 2021);
+        Ok(())
+    }
+
+    #[test]
+    fn test_timestamp_month() -> IonResult<()> {
+        let timestamp = Timestamp::with_year(2021).with_month(2).build()?;
+        assert_eq!(timestamp.month(), 2);
+        Ok(())
+    }
+
+    #[test]
+    fn test_timestamp_day() -> IonResult<()> {
+        let timestamp_1 = Timestamp::with_year(2021).with_month(2).build()?;
+        assert_eq!(timestamp_1.day(), 1);
+
+        let timestamp_2 = Timestamp::with_year(2021)
+            .with_month(2)
+            .with_day(4)
+            .build()?;
+
+        assert_eq!(timestamp_2.day(), 4);
+        Ok(())
+    }
+
+    #[test]
+    fn test_timestamp_hour() -> IonResult<()> {
+        let timestamp_1 =
+            Timestamp::with_ymd_hms(2021, 4, 6, 10, 15, 30).build_at_offset(-5 * 60)?;
+        assert_eq!(timestamp_1.hour(), 10);
+
+        let timestamp_2 =
+            Timestamp::with_ymd_hms(2021, 4, 6, 10, 15, 30).build_at_offset(5 * 60)?;
+        assert_eq!(timestamp_2.hour(), 10);
+
+        let timestamp_3 = Timestamp::with_ymd_hms(2021, 4, 6, 10, 15, 30).build_at_offset(0)?;
+        assert_eq!(timestamp_3.hour(), 10);
+        Ok(())
+    }
+
+    #[test]
+    fn test_timestamp_minute() -> IonResult<()> {
+        let timestamp_1 = Timestamp::with_ymd_hms(2021, 4, 6, 10, 15, 30).build_at_offset(-90)?;
+        assert_eq!(timestamp_1.minute(), 15);
+
+        let timestamp_2 =
+            Timestamp::with_ymd_hms(2021, 4, 6, 10, 15, 30).build_at_offset(-5 * 60)?;
+        assert_eq!(timestamp_2.minute(), 15);
+
+        let timestamp_3 =
+            Timestamp::with_ymd_hms(2021, 4, 6, 10, 15, 30).build_at_offset(5 * 60)?;
+        assert_eq!(timestamp_3.minute(), 15);
+
+        let timestamp_4 = Timestamp::with_ymd_hms(2021, 4, 6, 10, 15, 30).build_at_offset(0)?;
+        assert_eq!(timestamp_4.minute(), 15);
+        Ok(())
+    }
+
+    #[test]
+    fn test_timestamp_second() -> IonResult<()> {
+        let timestamp = Timestamp::with_ymd_hms(2021, 4, 6, 10, 15, 30).build_at_offset(-5 * 60)?;
+        assert_eq!(timestamp.second(), 30);
+        Ok(())
+    }
+
+    #[test]
+    fn test_timestamp_utc_hour() -> IonResult<()> {
```

Comment:
For each unit test, please include examples that test the unit's boundary across timezone offsets. (For example, in this `hour` test we see `5`/`10`/`15` -- we also need to verify the behavior when the result goes backwards or forwards across the day boundaries at `0`/`24`.)

Reply:
I'm unresolving this thread because I don't see tests that cross the `day` boundary. If the hour is `10`, what happens when the offset is `-11:00`? Or if it's `15` and the offset is `10:00`?